src/main/java/org/verdictdb/core/execplan/ExecutionTokenQueue.java

```diff
@@ -37,9 +38,14 @@ public ExecutionInfoToken poll() {
     return internalQueue.poll();
   }
 
-  public ExecutionInfoToken take() {
+  public synchronized ExecutionInfoToken take() {
     try {
-      return internalQueue.takeFirst();
+//      return internalQueue.takeFirst();
+      ExecutionInfoToken t = internalQueue.poll(5, TimeUnit.MINUTES);
```

Comment:
If timeout, it will throw an exception which will make circleci fail. I haven't observed the error yet. I will comment those lines and test multiple times on circleci.